### PR TITLE
UI/main screen design

### PR DESCRIPTION
**Pull request by AlecLeGrooker**
## Summary
This PR introduces the new design on the main screen to be more appealing and good-looking. Now the view all screen is not accessible anymore as there is a button on the screen to expand the list of workouts.

## Added Files
- Font.kt
- double_arrow.png
- dumbell_inner_shadow.png
- group.png
- quick_bodyweight.png
- quick_running.png
- quick_yoga.png
- running_innershadow.png
- yoga_innershadow.png
- contrail_one.ttf
- open_sans_bold.ttf
- open_sans_regular.ttf

## Modified Files
- EndToEndTest.kt
- MainScreenTest.kt
- MainScreen.kt
- ViewAllScreen.kt
- Color.kt
- trophy.png
- strings.xml

### MainScreen.kt
Modify the design to fit with the figma and use some parts of the previous view all screen to display some tabs when the workout list is expanded.
### Font.kt
Define two new fonts used in the main screen : OpenSans and ContrailOne
### double_arrow.png
PNG used as a button to expand the workout list.
### dumbell_inner_shadow.png
PNG displayed on the cards. It is imported because the inner shadow are too complicated in Kotlin so use an image is easier.
### running_innershadow.png
Same that dumbell but for running icon.
### yoga_innershadow.png
Same but for yoga icon.
### quick_bodyweight.png / quick_running.png / quick_yoga.png
Images used for the quick workout buttons.
### contrail_one.ttf / open_sans_bold.ttf / open_sans_regular.ttf 
Files defining the new fonts.
### EndToEndTest.kt
Modify the two test to fit with the new main screen architecture.
### MainScreenTest.kt
Modify previous tests to fit with the new architecture and also add some checks for the expand funcionality.
### ViewAllScreen.kt
Make the function to navigate to a workout overview public as now it is used in MainScreen.kt.
### Color.kt
Define some colours used in the main screen.
### trophy.png
Image used on the achievement button.
### strings.xml
Define some new strings used in the main screen.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt [hunk 1]
```diff
@@ -96,120 +125,225 @@ fun MainScreen(
...
-              .padding(vertical = 12.dp)
-              .testTag("ProfileSection")) {
+              .shadow(20.dp, clip = false)
+              .background(
+                  brush =
+                      Brush.linearGradient(
+                          colors =
+                              listOf(
+                                  DarkBlueTopBar1,
+                                  DarkBlueTopBar2), // Dégradé de DarkBlue à LightBlue
```
Comment (radutipurita) [inline app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt:143]: Maybe it would be a good idea to keep the comments in English.

**Review by radutipurita — COMMENTED** [review]
Thank you for this Pr, after checking also on the emulator the Ui of the Main Screen does indeed look nicer now. I had just one additional suggestion to make, maybe it would be better to make the titles My workouts, Quick Workout and My achievements bolder so they can stand out a bit more. Overall nice work, good job 👍

File: app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt [hunk 2]
```diff
@@ -96,120 +125,225 @@ fun MainScreen(
...
-              }
+            WorkoutCard(workoutItem, workoutViewModel, profile, navigationActions)
+            Spacer(modifier = Modifier.height(15.dp))
+          }
+        }
+      } else {
+        Text(
+            modifier = Modifier.fillMaxWidth().padding(12.dp).testTag("NoWorkoutMessage"),
+            textAlign = TextAlign.Center,
+            text = "No workouts yet",
```
[the remark this answers is not in the document]
Comment (AlecLeGrooker) [reply app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt:257]: Done

File: app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt (hunk 1 above)
Comment (AlecLeGrooker) [reply app/src/main/java/com/android/sample/ui/mainscreen/MainScreen.kt:143]: Resolved

**Comment by radutipurita** [comment]
Changes have been implemented. Approved.

**Review by radutipurita — APPROVED** [review]
Changes have been implemented. Approved.

**Pull request merged**